test/e2e/apiextensions/compositon_test.go

```diff
@@ -0,0 +1,365 @@
+												 "conditionType": "Ready",
+												 "conditionStatus": "False",
+												 "time": "0s"
+											  },
+											  {
+												 "conditionType": "Ready",
+												 "conditionStatus": "True",
+												 "time": "20s"
+											  },
+											  {
+												 "conditionType": "Synced",
+												 "conditionStatus": "False",
+												 "time": "0s"
+											  },
+											  {
+												 "conditionType": "Synced",
+												 "conditionStatus": "True",
+												 "time": "20s"
+											  }
+										   ]
+										},
+										"writeConnectionSecretToRef": {
+										   "namespace": "crossplane-system",
+										   "name": "nop-example-resource"
+										}
+									}
+								}`)},
+							},
+						},
+					},
+				}
+
+				if err := c.Create(ctx, comp); err != nil {
+					t.Fatalf("Create composition %q: %v", comp.GetName(), err)
+				}
+				t.Logf("Created composition %q", comp.GetName())
+
+				t.Cleanup(func() {
+					t.Logf("Cleaning up composition %q.", comp.GetName())
+					if err := c.Get(ctx, types.NamespacedName{Name: comp.GetName()}, comp); resource.IgnoreNotFound(err) != nil {
+						t.Fatalf("Get composition %q: %v", comp.GetName(), err)
+					}
+					if err := c.Delete(ctx, comp); resource.IgnoreNotFound(err) != nil {
+						t.Fatalf("Delete composition %q: %v", comp.GetName(), err)
+					}
+					t.Logf("Deleted composition %q", comp.GetName())
+				})
+
+				// nopRes := schema.GroupVersionResource{Group: "nop.example.org", Version: "v1alpha1", Resource: "nopresources"}
+
+				nopresource := &unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"apiVersion": "nop.example.org/v1alpha1",
+						"kind":       "NopResource",
+						"metadata": map[string]interface{}{
+							"name": "nop-example",
+						},
+						"spec": map[string]interface{}{
+							"coolField": "example",
+						},
+					},
+				}
+
+				if err := c.Create(ctx, nopresource); err != nil {
+					t.Fatalf("Create nop-example %q: %v", nopresource.GetName(), err)
+				}
+				// res, err := dc.Resource(nopRes).Namespace("default").Create(context.TODO(), nopresource, metav1.CreateOptions{})
+				if err != nil {
+					t.Fatalf("Create nop-example %q: %v", nopresource.GetName(), err)
+				}
+
+				t.Logf("Created nop-example %q", nopresource.GetName())
+
+				if err := wait.PollImmediate(10*time.Second, 90*time.Second, func() (done bool, err error) {
+					// if result, err := dc.Resource(nopRes).Namespace(res.GetNamespace()).Get(context.TODO(), res.GetName(), metav1.GetOptions{}); err != nil {
+					// 	return false, err
+					// }
+
+					// c.Get(context.TODO(), types.NamespacedName{Name: res.GetName(), Namespace: res.GetNamespace()}, res)
+
+					// TODO(rahulgrover99)
+					// Fetch condition of the resource and check if its ready or not
+
+					return true, nil
+				}); err != nil {
+					t.Errorf("nop-example %q never became Ready: %v", nopresource.GetName(), err)
+				}
```

Comment:
Figure some way out fetch the `READY` status of the resource.

Reply:
This could be a hint here in the unstructured deployments example: https://github.com/kubernetes/client-go/blob/master/examples/dynamic-create-update-delete-deployment/main.go#L177

It's using `unstructured.NestedInt64()` to probe down into the object to get the `spec.replicas` field, which it knows is a Int64.  For the Ready status, we could do something similar if they have `unstructured.NestedBool()` to probe down into the status.conditions path.

this is probably a bit trickier since conditions is a map or slice, so we'll need to grab the specific Ready status from inside there, but I think this approach from the link above is at least a good hint to guide us 🙏